examples/stable/test/property/ic_api/json/test/tests.ts

```diff
@@ -0,0 +1,146 @@
+import { Principal } from '@dfinity/principal';
+import { jsonStringify } from 'azle';
+import {
+    defaultPropTestParams,
+    expect,
+    getCanisterActor,
+    it,
+    Test
+} from 'azle/_internal/test';
+import fc from 'fast-check';
+
+import { _SERVICE as Actor } from './dfx_generated/canister/canister.did';
+
+// Helper function to create a Principal arbitrary
+const principalArb = fc
+    .array(fc.integer({ min: 0, max: 255 }), { minLength: 1, maxLength: 29 })
+    .map((bytes) => Principal.fromUint8Array(Uint8Array.from(bytes)));
+
+// Create an arbitrary for all the special types that jsonStringify can handle
+const specialValueArb = fc.oneof(
+    fc.constant(undefined),
+    fc.bigInt(),
+    principalArb,
+    fc.constant(NaN),
+    fc.constant(Infinity),
+    fc.constant(-Infinity),
+    fc.constant(-0),
+    fc.array(fc.integer()).map((arr) => new Int8Array(arr)),
+    fc.array(fc.integer()).map((arr) => new Int16Array(arr)),
+    fc.array(fc.integer()).map((arr) => new Int32Array(arr)),
+    fc.array(fc.integer({ min: 0, max: 100 })).map((arr) => {
+        try {
+            return new BigInt64Array(arr.map((n) => BigInt(n)));
+        } catch (_e) {
+            // Skip if conversion fails
+            return new BigInt64Array(0);
+        }
+    }),
+    fc.array(fc.integer()).map((arr) => new Uint8Array(arr)),
+    fc.array(fc.integer()).map((arr) => new Uint16Array(arr)),
+    fc.array(fc.integer()).map((arr) => new Uint32Array(arr)),
+    fc.array(fc.integer({ min: 0, max: 100 })).map((arr) => {
+        try {
+            return new BigUint64Array(arr.map((n) => BigInt(n)));
+        } catch (_e) {
+            // Skip if conversion fails
+            return new BigUint64Array(0);
+        }
+    }),
+    fc.array(fc.float()).map((arr) => new Float32Array(arr)),
+    fc.array(fc.float()).map((arr) => new Float64Array(arr)),
+    fc
+        .array(fc.tuple(fc.string(), fc.string()), { maxLength: 5 })
+        .map((entries) => new Map(entries)),
+    fc.array(fc.string(), { maxLength: 5 }).map((items) => new Set(items))
+);
+
+// Create an arbitrary for JSON-compatible values (strings, numbers, booleans, arrays, objects)
+const jsonCompatibleArb: fc.Arbitrary<unknown> = fc.letrec((tie) => ({
+    value: fc.oneof(
+        fc.string(),
+        fc.float(),
+        fc.boolean(),
+        fc.array(tie('value'), { maxLength: 2 }),
+        fc.dictionary(fc.string(), tie('value'), { maxKeys: 2 })
```

Comment:
I'd like more keys on the dictionary, maybe max of 20. Also I'd like any of the arbitraries that have nested values to be as absolutely arbitrary as possible. So Map, Set, and dictionary, I would like to use the final mixedObjectArb for the values.